wpengine/atlas-content-modeler
# [BH-1140] Show warning when trashing an entry that is referenced in a relationship field

- Adds a modal to warn people about the implications of trashing an entry linked to other entries via a relationship field.
- The modal appears when someone clicks the “Move to Trash” link in the Publish meta box on the post edit screen.
- Posts that aren't referenced in relationship fields do not display the modal.

This PR does not alter the “trash” links or “bulk trash” action on the entry listings admin page index. We could implement those in follow-up PRs or defer until later.

https://wpengine.atlassian.net/browse/BH-1140

## Testing

Includes e2e tests for the modal functionality. These also indirectly cover `has_relationship_references()`.

### To test manually
1. Create two models, one with a relationship field that refers to the other.
2. Create an entry for each model, one that refers to the other in the relationship field.
3. Visit the entry that is used in the relationship field.
4. Click “Move to Trash” in the publish meta box.
5. You should see a modal to confirm trashing the entry.

## Screenshots

<img width="326" alt="Screenshot 2021-09-07 at 10 42 42" src="https://user-images.githubusercontent.com/647669/132313934-4b57d1fc-d8bd-4c8d-8d84-0cfc1f14543d.png">

<img width="676" alt="Screenshot 2021-09-07 at 10 41 24" src="https://user-images.githubusercontent.com/647669/132313673-45894aa1-dd02-44af-baa6-e1dd35246f41.png">

## Documentation Changes

n/a

## Dependant PRs

Targets #222 but can be merged in any order.
